.changeset/cold-chefs-attend.md

```diff
@@ -0,0 +1,19 @@
+---
+'skeleton': patch
+'@shopify/hydrogen': patch
+'@shopify/cli-hydrogen': patch
+---
+
+✨ Change how customerAccountClient handle unauthorized customer during query/mutate
+
+When an unauthorized customer is presented during a query or mutate call. The customer will automatically be redirect to the login page and redirect back to current page when auth is complete.
+
+`customerAccount.handleUnauthorized()` is also added to run login check and redirect customer if unauthorized.
+
+---
+
+A few opt out is provided if this default behaviour is not ideal.
+
+1. pass `loginUrl` during `createCustomerAccountClient` to change the default login page from `/account/login` to something else.
+1. pass `unauthorizedHandler` during `createCustomerAccountClient` to change auto redirect all together
+1. run `isLoggedIn` ahead of `unauthorizedHandler`, `query` or `mutate` to handle one off unauthorized behaviour
```

Comment:
I've used the existing function names although Michelle and I talked about alternatives. @michenly I'll leave it to you to update the function names as needed!

Below is my attempt to lay out the key points here — thoughts?

```suggestion
There are two ways to override the default behavior for logged-out customers, depending on the functionality you need.

- If you want to customize logged-out redirect behavior across your entire application, then pass an `unauthorizedHandler` function to when initializing with `createCustomerAccountClient`.
- If you want to customize the redirect behavior on a case-by-base basis, anywhere in your app, run the `isLoggedIn` function to check whether the customer is logged in. This `isLoggedIn` check must be run before making any `query`, `mutate`, or `unauthorizedHandler` calls.
```